File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class AttackFrequency(BaseModel):
    attack_type: str
    count: int


class IncidentTrend(BaseModel):
    date: str
    count: int


class ResponseTimeBySeverity(BaseModel):
    severity: str
    avg_minutes: float


class IncidentsByGroup(BaseModel):
    name: str
    count: int


class HighRiskSystem(BaseModel):
    system_name: str
    count: int
    latest_severity: str


class RepeatedAttack(BaseModel):
    attack_type: str
    system_name: str
    count: int

File: frontend/src/types/analytics.ts
export interface AttackFrequency {
  attack_type: string;
  count: number;
}

export interface IncidentTrend {
  date: string;
  count: number;
}

export interface ResponseTimeBySeverity {
  severity: string;
  avg_minutes: number;
}

export interface IncidentsByGroup {
  name: string;
  count: number;
}

export interface HighRiskSystem {
  system_name: string;
  count: number;
  latest_severity: string;
}

export interface RepeatedAttack {
  attack_type: string;
  system_name: string;
  count: number;
}

export interface AlertLog {
  alert_id: number;
  rule_id: number;
  rule_name?: string;
  rule_type?: string;
  incident_id: number;
  triggered_at: string;
  message: string | null;
  acknowledged: boolean;
}

export interface AlertCount {
  count: number;
}
